Sources/PhoenixLiveViewNative/NavigationCoordinator.swift

```diff
@@ -16,7 +16,9 @@ class NavigationCoordinator: NSObject, UINavigationControllerDelegate, Observabl
             // while we're animating, don't update the currentRect otherwise it starts animating back
             // to the now-offscreen position of the source
             if case .idle = state {
-                currentRect = sourceRect
+                DispatchQueue.main.async {
+                    self.currentRect = self.sourceRect
+                }
```

Comment:
This is unnecessary. The `sourceRect` is only set from the button handler of `PhxModernNavigationLink`, which is always invoked on the main thread.